### ISSUE TITLE
Response missing `link` property for Requests compatibility

### ISSUE THREAD

**Issue by euri10** [encode/httpx#205]
in requests you get that very useful goodie 

https://github.com/psf/requests/blob/bedd9284c9646e50c10b3defdf519d4ba479e2c7/requests/models.py#L899-L915

that gives you parsed headers links

https://2.python-requests.org/en/master/user/advanced/#id14

I could not find it in httpx, but maybe I overlooked something

happy to send a PR copy-pasting what requests does if it's not available!

**Comment by sethmlarson** [comment]
This is definitely something that can be added, are there any other properties on the requests Response class that we're missing?
Instead of copy-pasting from Requests let's re-implement to suit our code. :)

**Comment by behrtam** [comment]
But does "A requests-compatible API." mean full 100% drop-in compatible?

A simple diff looks like this:
```python
>>> rx = httpx.get(url=url)
>>> rr = requests.get(url=url)
>>> [prop for prop in dir(rr) if not prop in dir(rx) and not prop.startswith('_')]
['connection', 'elapsed', 'is_permanent_redirect', 'iter_content', 'iter_lines', 'links', 'ok', 'reason']
>>>
```

**Comment by sethmlarson** [comment]
I don't think that's @tomchristie's intent to be 100% compatible. Maybe I'll let him decide whether we want this feature.

**Comment by behrtam** [comment]
Interesting question what the right balance might be ... is `is_redirect` already too much or should even `is_permanent_redirect` be there. So let's see.

**Comment by tomchristie** [comment]
Yeah, so the intent will be "broadly API compatible". Ie. we've got a small enough set of API differences that we can have a section that thoroughly documents where we differ from any existing `requests` API that's detailed in either of their "Getting Started" or "Advanced Usage" docs.

On a quick pass over:

* `.iterate_content`, `.iterate_lines` probably won't make the cut - since we'll want end up with slightly differing streaming bytes and text APIs, given our tighter typing.
* `.reason` - We currently have `reason_phrase`. I'd *probably* prefer our naming. Would *possibly* accept a synonym.
* `.connection` - Nope. Couldn't be API compatible even if we did expose something here.
* `.elapsed` - Probably okay, but wouldn't consider it a blocker.
* `.is_redirect` - Totally okay with current status of us having that in.
* `.is_permanent_redirect` -  Probably okay, but wouldn't consider it a blocker.
* `.ok` - Perhaps, but a bit meh towards the ambiguous naming.
* `.links` - Probably okay, but wouldn't consider it a blocker.
